### ISSUE TITLE
missing-parameter false positive when a table literal is passed to pcall()

### ISSUE THREAD

**Issue by ligurio** [EmmyLuaLs/emmylua-analyzer-rust#1250]
Imagine a file `pcall_table.lua` with the following Lua chunk:

```lua
local ok = pcall(function() end, {1, 2, 3})
print(ok)
```

and the following config `.emmyrc.json` for the emmylua_check: `{ "runtime": { "version": "LuaJIT" } }`.

Run the command: `emmylua_check --config .emmyrc.json pcall_table.lua`
The command reports:

```
---  [1 warning]
warning: expected 4 parameters but found 2. missing parameter: var2 
 missing parameter: var3 [missing-parameter]
  -->: :16:43

  15 │ 
  16 │ local ok = pcall(function() end, {1, 2, 3})
  17 │ print(ok)



Summary
  1 warning

Check completed with warnings
Check finished
```

It is expected no diagnostics.

The function `pcall()` is declared in `resources/std/global.lua` as

```
--- @generic T, R
--- @param f   sync fun(...: T...): R...
--- @param ... T...
```

A table literal argument (`{1, 2, 3}`) is treated as a tuple, so `T...` is expanded into three parameters `var1`, `var2`, `var3` (see `semantic/generic/instantiate_type/mod.rs`), and `pcall()` then expects 4 parameters.

Notes: `pcall(print, {1, 2, 3})` does not trigger it (variadic `print()` absorbs args), while `pcall(function() end, {1, 2, 3})` does.

Version: emmylua_check 0.24.0
